src/test/clojure/clara/performance/test_compilation.clj

```diff
@@ -0,0 +1,154 @@
+(ns clara.performance.test-compilation
+  (:require [clojure.test :refer :all]
+            [clara.rules.compiler :as com]
+            [clara.rules :as r]
+            [clara.rules.durability.fressian :as dura])
+  (:import [java.io ByteArrayOutputStream ByteArrayInputStream]))
+
+(defn filter-fn
+  [seed-keyword]
+  (= seed-keyword (keyword (gensym))))
+
+(defn time-execution
+  [func]
+  (let [start (System/currentTimeMillis)
+        _ (func)
+        stop (System/currentTimeMillis)]
+    (- stop start)))
+
+(defn execute-tests
+  [func iterations]
+  (let [execution-times (for [_ (range iterations)]
+                          (time-execution func))
+        sum #(reduce + %)
+        mean (/ (sum execution-times) iterations)
+        std (->
+              (into []
+                    (comp
+                      (map #(- % mean))
+                      (map #(Math/pow (double %) 2.0)))
+                    execution-times)
+              sum
+              (/ iterations)
+              Math/sqrt)]
+    {:std (double std)
+     :mean (double mean)}))
+
+(defn run-performance-test
+  [form]
+  (let [{:keys [description func iterations mean-assertion]} form
+        {:keys [std mean]} (execute-tests func iterations)]
+    (println (str \newline "Running Performance tests for:"))
+    (println description)
+    (println "==========================================")
+    (println (str "Mean: " mean "ms"))
+    (println (str "Standard Deviation : " std "ms" \newline))
+    (is (mean-assertion mean))))
+
+(def base-production
+  {:ns-name (symbol (str *ns*))})
+
+(defn generate-filter-productions
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [next-fact (symbol (str seed-sym "prime"))
+                    production (assoc base-production
+                                 :lhs [{:type (keyword seed-sym)
+                                        :constraints ['(= this ?binding) `(filter-fn ~'this)]}]
```

Comment:
Use syntax quote so that `=` is expanded. Even though Clara will accept the bare `=` symbol in many places, I think it is a bad practice due to it being able to be shadowed. Clara should be using `clojure.core/=`, not a bare `=`.
```clj
`(= ~'this ~'?binding)
```

This applies elsewhere in this ns too.